eng/testing/runsettings.targets

```diff
@@ -26,6 +26,9 @@
   <Target Name="GenerateRunSettingsFile"
           DependsOnTargets="SetupCoverageFilter">
```

Comment:
I think you could change this to an AfterTargets to not require the empty target.

Reply:
I added a `GenerateRunSettingsFileDependsOn` that conditionally includes `SetupCoverageFilter` so that it gets run before. I think that better preserves the intent, since with `AfterTargets`, I think I'd need to explicitly chain in `SetupCoverageFilter` to `PrepareForRun` so that it actually gets run.